src/isl/isl_type.rs

```diff
@@ -122,32 +57,35 @@ impl NamedIslType {
         let mut constraints = vec![];
         let annotations: Vec<&OwnedSymbolToken> = ion.annotations().collect();
 
-        // For named types check if it has the `type::` annotation
-        if !annotations.contains(&&text_token("type")) {
-            return Err(invalid_schema_error_raw(
-                "Named types must have `type::` annotation in their definition",
-            ));
-        }
+        let contains_annotations = annotations.contains(&&text_token("type"));
 
         let ion_struct = try_to!(ion.as_struct());
 
         // parses the name of the type specified by schema
-        let type_name: String = match ion_struct.get("name") {
+        let type_name: Option<String> = match ion_struct.get("name") {
             Some(name_element) => match name_element.as_str() {
-                Some(name) => name.to_owned(),
+                Some(name) => Some(name.to_owned()),
                 None => {
                     return Err(invalid_schema_error_raw(
                         "type names must be a string or a symbol with defined text",
                     ))
                 }
             },
-            None => {
-                return Err(invalid_schema_error_raw(
-                    "Named type must have a name field in its definition",
-                ))
-            } // If a named type doesn't have name field throw an error
+            None => None, // If there is no name field then it is an anonymous type
         };
 
+        if contains_annotations && type_name.is_none() {
+            // If a named type doesn't have name field throw an error
```

Comment:
A nit for some of these comments: Rust doesn't have a `throw` mechanism.
```suggestion
            // If a named type doesn't have name field return an error
```